qutebrowser/browser/shared.py

```diff
@@ -551,7 +560,7 @@ def _validated_selected_files(
                     f"Expected folder but got file, ignoring '{selected_file}'"
                 )
                 continue
-        else:
+        if qb_mode != FileSelectionMode.download:
```

Comment:
For starters, this triggers now when `qb_mode` is any of the above modes, i.e. the logic is completely different than it was before.

Reply:
Hum, yes. I will do tests and I will review a bit closer the code... Thank you for this insight. I am a bit intimidated by the complexity of the tests, I am not familiar with these things, but I will try to fix that...